<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=gbk">
    <title>PKA:positional K-mer Analysis</title>
    <style>
input[type="submit"]{
/* change these properties to whatever you want */
background-color: #555;
color: #fff;
border-radius: 10px;
font-size:1em;
height:30px;

}

code {
color: SteelBlue  ;
font-size:0.9em;
}


h2 {
font-family: Verdana;
color: SteelBlue  ;
}
h3 {
font-family: Verdana;
color: SteelBlue  ;
}
h4 {
font-family: Verdana;
color: SteelBlue  ;
}

td {
    vertical-align: top;
}

a:link {
    text-decoration: none;
	color: SlateBlue;
}

  </style>
    <style>
input[type="file"]{
/* change these properties to whatever you want */
background-color: #555;
color: #fff;
border-radius: 10px;
font-size:1em;
}
  </style>
    <style>
input[type="text"]{
/* change these properties to whatever you want */
background-color: #aaa;
color: #fff;
}
  </style>
  </head>
  <body style="margin:0 auto; width: 970px; text-align: center">
    <div style="font-family: Helvetica,Arial,sans-serif;text-align:
      left;" id="title"> <br>
      <h1><span class="title"> PKA: Positional K-mer Analysis</span><br>
      </h1>
    </div>
    <div style=" border-top: 8px solid DimGray; width: 970px"></div>
    <div style=" border-top: 8px solid White; width: 970px"></div>
    <table style="text-align: left; width: 436px; height: 32px;
      margin-left: auto; margin-right: 0px;" border="0" cellpadding="0"
      cellspacing="0">
      <tbody>
        <tr>
          <td style="vertical-align: top; width: 69px;">
            <form action="./index.html"> <input value="Home"
                type="submit"></form>
          </td>
          <td style="vertical-align: top; width: 69px;">
            <form action="./submit.html"> <input value="Server"
                type="submit"></form>
          </td>
          <td style="vertical-align: top; width: 0px;">
            <form action="./manual.html#install-pka-locally"> <input
                value="Code" type="submit"></form>
          </td>
          <td style="vertical-align: top; width: 69px;">
            <form action="./manual.html"> <input value="Manual"
                style="color: GreenYellow" type="submit"></form>
          </td>
          <td style="vertical-align: top; width: 69px;">
            <form action="./manual.html#examples"> <input
                value="Examples" type="submit"></form>
          </td>
          <td style="vertical-align: top; width: 0px;">
            <form action="./feedback.html"> <input value="Feedback"
                type="submit"></form>
          </td>
        </tr>
      </tbody>
    </table>
    <div id="TOC" style="font-family: Helvetica,Arial,sans-serif;
      text-align: left;"><span style="font-weight: bold;"></span><br>
      <ul id="mozToc">
        <!--mozToc h1 1 h2 2 h3 3 h4 4 h5 5 h6 6-->
        <li><a href="#What-is-PKA">What is PKA </a></li>
        <li><a href="#unique-featues-of-pka">Unique features of PKA </a></li>
        <li><a href="#using-pka">Using PKA</a>
          <ul>
            <li><a href="#web-server">Web server</a>
              <ul>
                <li><a href="#usage">Usage</a></li>
                <li><a href="#limitations">Limitations</a></li>
              </ul>
            </li>
            <li><a href="#install-pka-locally">Install PKA locally</a>
              <ul>
                <li><a href="#requirements">Requirements</a></li>
                <li><a
href="#download">Download</a></li>
                <li><a href="#installation">Installation</a></li>
                <li><a href="#test">Test</a></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><a href="#how-pka-works">How PKA works</a>
          <ul>
            <li><a href="#positional-kmer">Positional K-mer with shift and degeneracy</a>
              <ul>
                <li><a href="#positional-shift">Positional shift (<font color=Red>experimental</font>)</a></li></
                <li><a href="#degenerate-letter">Degenerate letter</a></li>
              </ul>
            </li>
            <li><a href="#statistical-models">Statistical models</a>
              <ul>
                <li><a href="#weighted">Weighted: two-sample t-test</a></li>
                <li><a href="#ranked">Ranked: rank-sum test</a></li>
                <li><a href="#unweighted">Unranked (default): binomial test</a>
                  <ul>
                    <li><a href="#background-sequences">Background sequences</a></li>
                    <li><a href="#shuffle-input-sequences">Shuffle input sequences</a></li>
                    <li><a href="#markov-model-from-input">Markov model from input sequences</a></li>
                    <li><a href="#markov-model-from-background">Markov model from background sequences</a></li>
                    <li><a href="#average-across-all-positions">Average across all positions (default)</a></li>
                    <li><a href="#other-parameters">Other parameters</a></li>

                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><a href="#input">Input</a>
          <ul>
            <li><a href="#input-type">Sequence type</a></li>
                <ul>
                <li><a href="#input-type-dna">DNA/RNA</a></li>
                <li><a href="#input-type-protein">Protein/Other</a></li>
                </ul>
			<li><a href="#input-format">Format</a></li>
				<ul>
				<li><a href="#tabular">Tabular / Column file</a></li>
				<li><a href="#fasta">FASTA</a></li>
				</ul>
            <li><a href="#preprocessing">Preprocessing</a></li>
              <ul>
                <li><a href="#default-formating">Default formating</a></li>
                <li><a href="#subregion">Analyze a sub-region</a></li>
                <li><a href="#filter-by-kmer">Filter by positional kmer</a></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><a href="#output">Output</a>
          <ul>
            <li><a href="#logo-plots">Logo plots</a></li>
            <li><a href="#tabular-data-file">Tabular data file</a></li>
          </ul>
        </li>
        <li><a href="#examples">Examples</a>
          <ul>
            <li><a href="#example1">Unranked: microRNA precursor / Drosha specificity</a></li>
            <li><a href="#example2">Weighted/Ranked: CRISPR/Cas9 cutting efficiency</a></li>
          </ul></li>
        <li><a href="#command-line-options">Command Line Options</a></li>
        <li><a href="#copyright">Copyright</a></li>
        <li><a href="#author">Author</a></li>
        <li><a href="#acknowlegements">Acknowledgements</a></li>
        <li><a href="#reference">Reference</a></li>
      </ul>
      <br>
    </div>
    <div style="font-family: Helvetica,Arial,sans-serif; text-align:
      left;">
      <h1><a class="mozTocH1" name="What-is-PKA"></a>What is PKA </h1>
      <font color="Blue">Positional K-mer Analysis (PKA)</font> is a
      computational tool for the identification and visualization of
      positionally enriched or depleted k-mers (subsequences of length
      k)
      from a set of aligned sequences, typically nucleic acids (DNA or
      RNA)
      or amino acids (protein).
      <h1><a class="mozTocH1" name="unique-featues-of-pka"></a>Unique
        features of
        PKA </h1>
      Compare to existing sequence logo tools, the two most prominant
      features of PKA is the detection and visualization of motifs
      longer
      than one letters and the incorporation of weights for each input
      sequence.
      <br>
      <br>
      PKA combines motif discovery with visualization. Compared to
      existing
      sequence logo generating tools, PKA is unique in several ways:<br>
      1. visualize not only single nucleotides but also short motifs,
      i.e. K-mer logo<br>
      2. plot frequency logo, information content logo, probability
      logo, and K-mer logo in one run<br>
      3. can generate logos from ranked or weighted sequences<br>
      4. allow positional shifts<br>
      5. allow degenerate nucleotides<br>
      6. logos can be generated with colorblind friendly colors<br>
      7. allow users to score new sequenes using the learned positional
      sequence features <br>
      <h1><a class="mozTocH1" name="using-pka"></a>Using PKA</h1>
      PKA can be accessed as an online tool at: <a
        href="http://pka.wi.mit.edu">http://pka.wi.mit.edu</a>. In
      addition,
      the C++ source code as well as the php/html web code can be
      downloaded
      and installed locally. <br>
      <h2><a class="mozTocH2" name="web-server"></a>Web server</h2>
      To use PKA online, please visit <a href="http://pka.wi.mit.edu/">http://pka.wi.mit.edu</a><br>
      <h3><a class="mozTocH3" name="usage"></a>Usage</h3>
      Users paste or upload sequences to the server and the job will run
      on
      the background. Once submitted, users are redirected to the result
      page
      which displays PKA screen output and refreshes every second. Users
      are
      encouraged to provide an email address when submitting the job. If
      provided, two emails will be sent, one immediately after
      submitting the
      job, and the other after results are available. Both emails
      contain the
      link to the result page. <br>
      <br>
      Note that user data will be <span style="color: red;">removed 72
        hours</span>
      after the job is finished. Please download the data as soon as it
      is
      available. It is recommended user to provide an email address to
      receive a notification.<br>
      <br>
      <h3><a class="mozTocH3" name="limitations"></a>Limitations</h3>
      The web server implements most functionality of PKA, and should be
      sufficient for most users. However, for simplicity and to avoid
      excess
      demand of server resources, a few restricitons have been imposed
      on the
      web version of PKA:<br>
      <ol>
        <li><span style="font-weight: bold;">kmer length</span>: the max
          length of kmer is limited to 6.<br>
        </li>
        <li><span style="font-weight: bold;">sequence shuffling</span>:
          if
          choose to shuffle sequence to generate the
          background,&nbsp; one can only preserve upto 4-letter
          frequency. <br>
        </li>
        <li><span style="font-weight: bold;">model prediction</span>:
          the binary version allows user to score a new
          set of sequences using the sequence bias learned from the
          input. The
          web version does not support this functionality at this
          moment.<br>
        </li>
      </ol>
      <h2><a class="mozTocH2" name="install-pka-locally"></a>Install PKA
        locally</h2>
      <p>PKA is a open-source project and users can download the source
        code, compile and install locally to enable all functionality.</p>
      <h3><a class="mozTocH3" name="requirements"></a>Requirements</h3>
      PKA has been successfully tested on Linux and Mac.<br>
      <br>
      <span style="font-weight: bold;">To compile: GCC</span><br>
      PKA is written in C++ and reuses some libraries from <a
        href="http://www.boost.org/">Boost</a> version 1.57.0. To
      compile PKA
      locally, one should have a recent version of <a
        href="https://gcc.gnu.org/">GCC</a> that is compatible with the
      g++
      option <code>-std=c++11</code>.
      <br>
      <br>
      <span style="font-weight: bold;">To generate PNG and PDF files:
        ImageMagick &amp; Ghostscript</span><br>
      Logos are generated in postscript (ps) files and then converted to
      PNG
      using the program <span style="color: rgb(51, 51, 255);">convert</span>
      in package <span style="color: rgb(51, 51, 255);">ImageMagick</span>.
      In addition, <span style="color: rgb(51, 51, 255);">Ghostscript</span>
      is used to convert postscript files to PDF files.<br>
      <br>
      <h3><a class="mozTocH3" name="download"></a>Download</h3>
      The PKA source code (including web code) is hosted on GitHub: <a
        href="https://github.com/xuebingwu/PKA">https://github.com/xuebingwu/PKA</a><br>
      <br>
      To clone the latest (may not be stable) version:<br>
      <br>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Monaco;"><small>git
            clone
            https://github.com/xuebingwu/PKA.git</small></div>
      <br>
      <br>
      To download stable releases of PKA: <a
        href="https://github.com/xuebingwu/PKA/releases/">https://github.com/xuebingwu/PKA/releases/
      </a><br>
      <br>
      <h3><a class="mozTocH3" name="installation"></a>Installation</h3>
      <br>
      Once download finished, Unzip if necessary, enter the subfolder
      <code>src</code>,
      then type <code>make</code>: <br>
      <br>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;"><small><span
              style="font-family: Monaco;">cd src</span><br
              style="font-family: Monaco;">
            <span style="font-family: Monaco;">make</span></small>
          <br>
        </div>
      <br>
      The compiled programs are in the folder <code>bin</code>.<br>
      To recompile, type <code>make clean</code> then <code>make</code>.<br>
      <br>
      <h3><a class="mozTocH3" name="test"></a>Test</h3>
      Go to the folder <code>bin</code>, and run the following examples:<br>
      <br>
      1. Unweighted sequences in fasta, default background (relative to
      other
      positions)<br>
      <br>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Monaco;"><small>./PKA
          ../test/input/input-fixed-length.fa -o
          ../test/output/unweighted-default</small></div>
      <br>
      2. Weighted sequences<br>
      <br>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Courier New,Courier,monospace;"><small><span
            style="font-family: Monaco;">./PKA
            ../test/input/input.weighted.txt
            -weighted -o ../test/output/weighted</span> </small></div>
      <br>
      If PKA works then the screen should output no errors. Please refer
      to
      the manual for other examples and explanation of the output files.<br>
      <br>
 
     <h1><a class="mozTocH1" name="how-pka-works"></a>How PKA works</h1>

	  <p>Below is a detailed description of PKA's concept and statistical models.</p>

      <h2><a class="mozTocH2" name="positional-kmer"></a>Positional k-mer with shift and degeneracy</h2>

		<p> A positional k-mer (or simply kmer in this document) is a short sequence of length k associated with a specific position. Throughout this document, a positional kmer has the format of <code>sequence:position:shift</code>. For example, most introns start with a GU dinucleotide, and this dinucleotide motif (k=2) can be represented as <code>GU:1:0</code>, i.e. GU at position 1 with no shift. The position of the first letter is used as the position of the kmer. </p>
		
        <h3><a class="mozTocH3" name="positional-shift"></a>Positional shift (<font color=Red>experimental</font>)</h3>

  In many other cases, sequence motifs are well positioned but also allow some flexibility, and this is modeled by the positional shift. For example, the canonical polyadenylation signal, AAUAAA, typically locates about 18 to 23 nucleotides upstream of the cleavage site, can be represented as <code>AAUAAA:-23:5</code>, i.e. the hexamer motif can be found at position -23, allowing a shift of 5, which means the motif can also occur at 5 other positions: -22, -21, -20, -19, and -18. In PKA, the position of a kmer is defined as the position of the first letter in the kmer, and positional shift is only allowed downstream of the start site, i.e. shift to the right. Positional shift is an experimental feature still under development. For example currently it is not fully supported by some background models (See below).   
        

      <h3><a class="mozTocH3" name="degenerate-letter"></a>Degenerate
        letter</h3>
		
		Biological sequence signals can also tolerate some mismatches, leading to degeneracy in the motif. For example, in addition to AAUAAA, AUUAAA is also used as a polyadenylatoin signal. Therefore a more general form of the polyadenylation signal is <code>AWUAAA:-23:5</code>, where W is either A or U. PKA supports all degenerate codes defined in the <a
        href="http://en.wikipedia.org/wiki/Nucleic_acid_notation">IUPAC code</a> (see table below). <br>
      <br>
      In default mode, PKA does not report kmers with degenerate letters. To enable kmer with degeneracy, use <code>-degenerate</code> to specify
      all allowed letters, <span style="color: red;">including
        non-degenerate letters</span>. For example, to include R and Y
      in addition to ACGT in DNA sequences, use <code>-degenerate ACGTRY</code> instead of <code>-degenerate RY</code>. <br>
      <br>
      Two special cases: 1) to allow all IUPAC letters, use <code>-degenerate all</code>. 2) to
      allow kmers with gaps, use <code>-degenerate ACGTN</code> or simply <code>-gapped</code>. Note that the letter <code>N</code> can not be the first letter or the last letter.<br>
      <br>
      <table style="width: 445px; height: 470px;" border="1"
        cellpadding="2" cellspacing="2">
        <tbody>
          <tr>
            <td style="font-weight: bold; width: 60px; text-align:
              center;">Symbol </td>
            <td style="font-weight: bold; width: 200px; text-align:
              center;">Description </td>
            <td style="font-weight: bold; width: 25px; text-align:
              center;" colspan="5" rowspan="1">Bases represented</td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">A </td>
            <td style="width: 200px; text-align: center;">Adenine </td>
            <td style="width: 25px; text-align: center;">A </td>
            <td style="width: 25px; text-align: center;"><br>
            </td>
            <td style="vertical-align: top; width: 25px; text-align:
              center;"><br>
            </td>
            <td style="vertical-align: top; text-align: center;"><br>
            </td>
            <td colspan="1" rowspan="5" style="vertical-align: middle;
              text-align: center;">1<br>
            </td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">C </td>
            <td style="width: 200px; text-align: center;">Cytosine </td>
            <td style="width: 25px; text-align: center;"><br>
            </td>
            <td style="vertical-align: top; width: 25px; text-align:
              center;">C<br>
            </td>
            <td style="vertical-align: top; width: 25px; text-align:
              center;"><br>
            </td>
            <td style="vertical-align: top; text-align: center;"><br>
            </td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">G </td>
            <td style="width: 200px; text-align: center;">Guanine </td>
            <td style="width: 25px; text-align: center;"><br>
            </td>
            <td style="vertical-align: top; width: 25px; text-align:
              center;"><br>
            </td>
            <td style="vertical-align: top; width: 25px; text-align:
              center;">G<br>
            </td>
            <td style="vertical-align: top; text-align: center;"><br>
            </td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">T </td>
            <td style="width: 200px; text-align: center;">Thymine </td>
            <td style="width: 25px; text-align: center;"><br>
            </td>
            <td style="vertical-align: top; width: 25px; text-align:
              center;"><br>
            </td>
            <td style="vertical-align: top; width: 25px; text-align:
              center;"><br>
            </td>
            <td style="vertical-align: top; text-align: center;">T<br>
            </td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">U </td>
            <td style="width: 200px; text-align: center;">Uracil </td>
            <td style="width: 25px; text-align: center;"><br>
            </td>
            <td style="vertical-align: top; width: 25px; text-align:
              center;"><br>
            </td>
            <td style="vertical-align: top; width: 25px; text-align:
              center;"><br>
            </td>
            <td style="vertical-align: top; text-align: center;">U<br>
            </td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">W </td>
            <td style="width: 200px; text-align: center;">Weak </td>
            <td style="width: 25px; text-align: center;">A </td>
            <td style="width: 25px; text-align: center;"> <br>
            </td>
            <td style="width: 25px; text-align: center;"><br>
            </td>
            <td style="vertical-align: top; text-align: center;">T<br>
            </td>
            <td colspan="1" rowspan="6" style="vertical-align: middle;
              text-align: center;">2<br>
            </td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">S </td>
            <td style="width: 200px; text-align: center;">Strong </td>
            <td style="width: 25px; text-align: center;"> <br>
            </td>
            <td style="width: 25px; text-align: center;">C<br>
            </td>
            <td style="vertical-align: top; width: 25px; text-align:
              center;">G<br>
            </td>
            <td style="vertical-align: top; text-align: center;"><br>
            </td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">M </td>
            <td style="width: 200px; text-align: center;">aMino </td>
            <td style="width: 25px; text-align: center;">A </td>
            <td style="width: 25px; text-align: center;">C</td>
            <td style="vertical-align: top; width: 25px; text-align:
              center;"><br>
            </td>
            <td style="vertical-align: top; text-align: center;"><br>
            </td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">K </td>
            <td style="width: 200px; text-align: center;">Keto </td>
            <td style="width: 25px; text-align: center;"> <br>
            </td>
            <td style="width: 25px; text-align: center;"><br>
            </td>
            <td style="vertical-align: top; width: 25px; text-align:
              center;">G<br>
            </td>
            <td style="vertical-align: top; text-align: center;">T<br>
            </td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">R </td>
            <td style="width: 200px; text-align: center;">puRine </td>
            <td style="width: 25px; text-align: center;">A </td>
            <td style="width: 25px; text-align: center;"><br>
            </td>
            <td style="vertical-align: top; width: 25px; text-align:
              center;">G<br>
            </td>
            <td style="vertical-align: top; text-align: center;"><br>
            </td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">Y </td>
            <td style="width: 200px; text-align: center;">pYrimidine </td>
            <td style="width: 25px; text-align: center;"> <br>
            </td>
            <td style="width: 25px; text-align: center;">C<br>
            </td>
            <td style="vertical-align: top; width: 25px; text-align:
              center;"><br>
            </td>
            <td style="vertical-align: top; text-align: center;">T<br>
            </td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">B </td>
            <td style="width: 200px; text-align: center;">not A (B comes
              after A)<br>
            </td>
            <td style="width: 25px; text-align: center;"> <br>
            </td>
            <td style="width: 25px; text-align: center;"> C<br>
            </td>
            <td style="width: 25px; text-align: center;"> G<br>
            </td>
            <td style="text-align: center;"> T<br>
            </td>
            <td style="text-align: center; vertical-align: middle;"
              colspan="1" rowspan="4">3</td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">D </td>
            <td style="width: 200px; text-align: center;">not C (D comes
              after C) <br>
            </td>
            <td style="width: 25px; text-align: center;"> A<br>
            </td>
            <td style="width: 25px; text-align: center;"> <br>
            </td>
            <td style="width: 25px; text-align: center;"> G<br>
            </td>
            <td style="text-align: center;"> T<br>
            </td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">H </td>
            <td style="width: 200px; text-align: center;">not G (H comes
              after G) <br>
            </td>
            <td style="width: 25px; text-align: center;"> A<br>
            </td>
            <td style="width: 25px; text-align: center;"> C<br>
            </td>
            <td style="width: 25px; text-align: center;"> <br>
            </td>
            <td style="text-align: center;"> T<br>
            </td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">V </td>
            <td style="width: 200px; text-align: center;">not T (V comes
              after T and U) <br>
            </td>
            <td style="width: 25px; text-align: center;"> A<br>
            </td>
            <td style="width: 25px; text-align: center;"> C<br>
            </td>
            <td style="width: 25px; text-align: center;"> G<br>
            </td>
            <td style="text-align: center;"> <br>
            </td>
          </tr>
          <tr>
            <td style="width: 60px; text-align: center;">N </td>
            <td style="width: 200px; text-align: center;">No idea <br>
            </td>
            <td style="width: 25px; text-align: center;"> A<br>
            </td>
            <td style="width: 25px; text-align: center;"> C<br>
            </td>
            <td style="width: 25px; text-align: center;"> G<br>
            </td>
            <td style="text-align: center;"> T<br>
            </td>
            <td style="text-align: center;">4 <br>
            </td>
          </tr>
        </tbody>
      </table>
      <br>
  



      <h2><a class="mozTocH2" name="statistical-models"></a>Statistical
        models</h2>

		<p>
      For a given value of k, PKA enumerates all possible kmers
      (allowing degenerate letters when specified), determine their presence /
      absence at every possible positions in all input sequences, and report their
      enrichment / depletion using statistical models specified by
      users.</p>

      <h3><a class="mozTocH3" name="weighted"></a>Weighted: two-sample t-test</h3>
      When turned on by <code>-weighted</code>,
PKA
      searches for kmers at certain positions whose presence / absence
      in
      a sequence is associated with higher / lower weight, respectively.
      For
      each kmer at each position, all input sequences are divided into
      two
      sets: <br>
      <br>
      1. positive sequences: sequences containing this kmer at this
      position,
      allowing position shift and degenerate letters, if specified.<br>
      2. negative sequences: all other sequences, i.e., sequences that
      do not contain this kmer at this position.<br>
      <br>
      A <code>one-sided two-sample Student's t test</code>
      is used to compare the weights of the two sets of sequences. The
      sign
      of the t statistics determines whether it is an enrichment
      (sequences
      with this kmer at this position have larger weight) or depletion.<br>
      <br>
      Note that the input sequences need not to be sorted by weight.<br>
      <h3><a class="mozTocH3" name="ranked"></a>Ranked: rank-sum test</h3>
      When
      turned on (by <code>-ranked</code>), PKA assumes the input sequences (FASTA
      or
      Tabular) are ranked, and enrichment / depletion of a kmer at a
      position
      means sequences with this kmer at this position tend to be ranked
      at
      the top / bottom, respectively. <br>
      <br>
      Similar to weighted sequences, ranked sequences are divided into a
      positive set and a negative set based on the presence / absence of
      a
      kmer of interest. Instead of Student's t-test, the nonparametric <code>Wilcoxon rank-sum test</code>,
      also called the <code>Mann-Whitney U test</code>,
      is used to determine the signficance of positional enrichment /
      depletion of a kmer. Assuming sample size larger than 8, the
      distribution of the U statistics is approximated by a Normal distribution. The resultant z-score and p-value from the Normal
      approximation is reported in the output. Again the sign of the
      z-score
      determines enrichment or depletion.<br>
      <br>
      Note that this model is activated by the option <code>-ranked</code>, which
      will
      ignore <code>-weight</code> if specified. Also note that sequences need to be
      sorted prior to PKA analysis. <br>
      <h3><a class="mozTocH3" name="unweighted"></a>Unranked (default): binomial test</h3>
      When neither <code>-ranked</code> nor <code>-weighted</code> is specified, PKA assumes
      the input is unweighted. PKA uses <code>one-sided Binomial test</code>
      to determine the significance of a kmer at a position.
      Specifically,
      the number of trials is the total nubmer of sequences, and the
      number
      of success is the number of sequences containing this kmer at this
      position. The hypothesized probability of success, i.e. the
      expected /
      background probability of a sequence containing this kmer at this
      position, can be calculated in several ways as specified by users
      (see
      details below). A z-score is also reported in the output, assuming
      Normal approximation. The sign of the z-score also indicates
      enrichment
      or depletion.<br>
      <br>
      The hypothesized probability of success can be calculated in five
      different ways:<br>
      <h4><a class="mozTocH4" name="background-sequences"></a>Background
        sequences</h4>
      <p>Users can provide a set of background sequences (<code>-bgfile</code>)
      that can be aligned in the same way. In this case, the expected
      probability of a sequence with a particular kmer at a particular
      position is estimated by the fraction of background sequences with
      this kmer at this position. </p>
      <p>
      The potential advantage of using carefully selected background
      sequences is that it may enrich desired signals by normalizing out
      known or even unknown sequence bias. However, a background
      sequence set may not be available or not large enough to for reliable
      estimation of the expected probability of success. Small set of background
      sequences lead to noisy output, especially for longer kmers. To avoid a zero estimation of expected probability, a pseudocount is added to kmer counts, which can be set using <code>-pseudo</code>. The defaut value of pseudocount is <code>1e-9</code>.</p>
      <h4><a class="mozTocH4" name="shuffle-input-sequences"></a>Shuffle
        input sequences</h4>
      This is implemented in the option <code>-shuffle n,m</code> (shuffle each sequence n times
      preserving m-letter frequency). PKA uses the <a
        href="http://digital.cs.usu.edu/%7Emjiang/ushuffle/">ushuffle</a>
      code to shuffle input sequences while preserving the frequency of
      all
      possible letter combinations of length specified by users. For
      example,
      in many genomic applications, DNA sequences are shuffled while
      preserving di-nucleotide frequences. Each sequence can be shuffled
      multiple times (default=100) to enable reliable estimation of the
      background probability of sequence containing the kmer at a
      particular
      position. However, this will lead to a linear increase in run
      time. In
      addition, for relatively short sequences, it is impossible to
      preserve
      even dinucleotide frequencies. Similar to using background sequences, pseudocount is used to avoid zero estimation of expected probability.<br>
      <h4><a class="mozTocH4" name="markov-model-from-input"></a>Markov
        model from input sequences</h4>
      This is activated by <code>-markov</code>
      specifying the order of the Markov model, which summarizes letter
      frequency and dependence between neighboring letters in input
      sequences. Currently PKA supports zero, first, and second order
      Markov
      models. The hypothesized probability of a sequence containing a
      particular kmer at a particular position is then determined by the Markov model.
      <br>
      <br>
      For example, the first-order Markov model learned from input
      sequences
      includes all mono- and di-letter frequencies, which directly serve
      as
      the hypothesized probability for kmers of length 1 and 2. For
      longer
      kmers, the hypothesized probability is calculated using the
      property of
      Markov models. For example, the probability of observing the kmer
      ABCD at a particular position is <br>
      <br>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Monaco;"><small>p(ABCD) = p(AB) x p(BC)/p(B) x
          p(CD)/p(C) = p(AB) x p(BC) x p(CD) / p(B) / p(C)</small> </div>
      <br>
      Where p(AB), p(BC), p(CD), p(B), and p(C) are the fraction of AB,
      BC,
      and CD di-letters and the fraction of the letter B and C in the
      input
      sequences, respectively.<br>
      <br>
      For zero-order Markov model:<br>
      <br>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Monaco;"><small>p(ABCD) = p(A) x p(B) x p(C) x p(D)</small>
      </div>
      <br>
      For second-order Markov model:<br>
      <br>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Monaco;"><small>p(ABCD) = p(ABC) x p(BCD) / p(BC)</small><br>
      </div>
      <br>
      The advantage over shuffling sequences is that it can be much
      faster
      and / or less noisy. However, at this moment Markov models do not
      accurately handle positional shifts and degenerate nucleotides
      (email
      authors for details), and does not support higher order Markov
      models. Please choose other background models when either
      positional shift or degenerate letters is used. <br>
      <h4><a class="mozTocH4" name="markov-model-from-background"></a>Markov
        model from background sequences</h4>
      This is activated when both <code>-markov</code> and <code>-bgfile</code> are specified. The only difference from <code>-markov</code> alone is that
      the Markov model is learned from background sequences specified by <code>-bgfile</code>.
      <h4><a class="mozTocH4" name="average-across-all-positions"></a>Average across all positions (default)<br>
      </h4>
      This is activated by default if none of the above model is
      specified (<code>-bgfile, -shuffle, -markov</code>).
      It is preferred because it is fast, less noisy, handles both
      positional
      shift and degenerate letters. In this model, the hypothesized
      probability of a sequence with a certain kmer at a certain
      position is
      the average observed success rate (fraction of input sequences
      containing this kmer at a given position) averaged over all
      positions. <br>
      <br>
	  	  
      <h4><a class="mozTocH4" name="other-parameters"></a>Other parameters<br>
      </h4>
      <b> Min count/fraction </b> (<code>-minCount</code>): minimum number or fraction of sequences to have this kmer to include in output. Default is 5. If a value between 0 and 1 is used, it will be interpreted as the minimum fraction. 
      <br><br>
      <b> Pseudo-count </b> (<code>-pseudo</code>): Pseudo-count added to background counts. default=1.0. Ignored by Markov models <code>-markov</code>
      <br><br>
      <b> P value cutoff </b> (<code>-pc</code>): Bonferroni corrected P value cut-off. Positions containing letters with a smaller P will be highlighted by red coordinates.
      <br><br>
	  
      <h1><a class="mozTocH1" name="input"></a>Input</h1>
      <h2><a class="mozTocH2" name="input-type"></a>Sequence type</h2>
		PKA is designed primarily for analyzing DNA or RNA sequences, but can also be used to study protein and other arbitrary types of sequences via the option <code>-alphabet</code>.
      <h3><a class="mozTocH3" name="input-type-dna"></a>DNA/RNA</h3>
	  When <code>-alphabet</code> is not specified by users, PKA use the default <code>-alphabet ACGT</code>. Note RNA will be converted to DNA by default so for RNA input there is no need to specify <code>-alphabet ACGU</code>. 
      <h3><a class="mozTocH3" name="input-type-protein"></a>Protein/Other</h3> 
	  <p> To run PKA on protein sequences, please set <code>-alphabet protein</code>, which is equivalent to <code>-alphabet ACDEFGHIJKLMNOPQRSTUVWY</code>. </p>
	  
	  <p> Similarly, to run PKA on other types of sequences, use <code>-alphabet </code> to specify all allowed letters.</p>

      <h2><a class="mozTocH2" name="input-format"></a>Format</h2>
      <a>PKA currently supports both FASTA format and Tabular format. <font
          color="red">Note that all sequences will be converted to upper
          case by default</font>. Lines start with <font color="blue">
          # </font> will be treated as comments and ignored.</a>
      <h3><a class="mozTocH3" name="tabular"></a>Tabular / Column file</h3>
      <a>PKA will treat <font color="red">ANY</font> of the following
        characters as delimiter: tab ("\t"), comma (","), vertical bar ("|"), and space (" ").
        <br>
        <br>
        Example 1: minimal ranked or unweighted tabular input <br>
        <br>
      </a>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Monaco;"><a><small>CGTAGCTAGCTACGTAGCTA<br>
            CGTACGTAGCTAGCTAGCTA<br>
            GCTGTGATCGTAGCTAGCTG<br>
            ...</small></a></div>
      <a><br>
        Example 2: minimal weighted input <br>
        <br>
      </a>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Monaco;"><a><small>
			CGTAGCTAGCTACGTAGCTA 3.4<br>
            CGTACGTAGCTAGCTAGCTA 8.7<br>
            GCTGTGATCGTAGCTAGCTG 1.2<br>
            ...</small></a></div>
      <br>
      <a>Example 3: additional columns in input (require <code>-seq 2 -weight
        3</code> in
        this example) <br>
      </a><a><br>
      </a>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Monaco;"><a><small>
			seq1 CGTAGCTAGCTACGTAGCTA 3.4<br>
            seq2 CGTACGGAGCTAGCTAGCTA 8.7<br>
            seq3 GCTGTGAGCGTAGCTAGCTG 1.2<br>
            ...</small></a></div>
      <a><br>
        Notes: <br>
        1. Each line is a sequence <br>
        2. Weights do not need to be sorted<br>
        3. Sequences do NOT need to be of the same length (see option
        <code>-region</code>)<br>
        4. If sequence is not in column 1 or weight not in column 2,
        then their
        position needs to be specified by <code>-seq</code> and <code>-weight</code>. For
        example
        <code>-seq 2 -weight 3</code> for example 3 above. </a><br>
      <br>
      <h3><a class="mozTocH3" name="fasta"></a>FASTA</h3>
      <a>Here is an example of a FASTA file: <br>
      </a><a><br>
      </a>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Monaco;"><a><small>&gt;seq1
            <br>
            CGTAGCTAGCTAGCTAGCTA<br>
            &gt;seq2<br>
            CGTAGCTAGCTAGCTAGCTA<br>
            &gt;seq3<br>
            CGTAGCTAGCTAGCTAGCTA<br>
            &gt;seq4<br>
            CGTAGCTAGCTAGCTAGCTA<br>
            ...</small></a></div>
      <a><br>
        Notes on FASTA file format: <br>
        1. Each sequence include an ID line and a sequence line <br>
        2. IDs always start with '&gt;' <br>
        3. IDs are not used by PKA and thus do not need to be unique <br>
        4. Sequences do not need to be of the same length (see option
        <code>-region</code>)<br>
        5. Can be treated as ranked (see option <code>-ranked</code>) but not
        weighted (need to use tabular format) </a>

      <h2><a class="mozTocH2" name="preprocessing"></a>Preprocessing</h2>

      <h3><a class="mozTocH3" name="default-formating"></a>Default formating</h3>
      <p>For DNA/RNA sequences (-alphabet ACGT or -alphabet ACGU), U
        will be converted to T</p>
      <p>All sequences will be converted to upper case.</p>
      <h3><a class="mozTocH3" name="subregion"></a>Analyze a sub-region</h3>
      <p>Use <code>-region</code> option to specify the sub-region to be
        analyzed. Specifically, <code>-region n1,n2</code> specifies two numbers, <code>n1</code>,
        and <code>n2</code>, where <code>n1</code> is the position of the first letter to be
        included, and <code>n2</code> is the position of the last letter to be
        included. For example, <code>-region 1,20</code> tells PKA to analyze the first
        20 letters. This option allows one to analyze sequences of
        different length. In addition, <code>n1</code> and <code>n2</code> can take non-positive
        integers, which PKA interpreted as the distance from the end.
        For example, the default setting <code>-region 1,0</code> means take the entire
        sequence without trimming. Another example is <code>-region -9,0</code>, which
        tells PKA to analyze the last 10 letters.</p>
      <h3><a class="mozTocH3" name="filter-by-kmer"></a>Filter by positional kmer</h3>
		<h4><a class="mozTocH4" name="filter-by-kmer-why"></a>Why</h4>
        <p> PKA allows user to filter input sequences by the presence (<code>-select</code>) or absence (<code>-remove</code>) of a list of positional km
ers. This feature enables one to study dependence between positional kmers (see examples ***).</p>
                <h4><a class="mozTocH4" name="filter-by-kmer-format"></a>Format</h4>
<p>A positional kmer is specified in the form of <strong>sequence:position:shift</strong>, such as GT:3:0, which means GT starting at position 3 with no shift. Note that sequence can contain degenerate DNA letters. Position coordinates start at 1
 for the first letter in sequence. </p>
        <p> If using multiple positional kmers, separate each by comma (,). For example, GT:3:0,ACT:11:0,GGG:45:0. Note that when multiple positional kmers specified, the presence of any positional kmer is defined as a match. <font color="red">DO NOT</font> leave a space before or after the comma! </p>
        <h4><a class="mozTocH4" name="filter-by-kmer-example"></a>Examples</h4>
        <p>Here are two more examples: </p>
<p><code>-select G:35:3,CNC:47:0</code> </p>
<p> Select (keep) sequences with G at position 35, 36, 37, or 38 (note shift=3), <strong>OR</strong> with CNC at position 47. Here N means any nucleotide, thus can be any of CAC, CGC, CTC, and CCC. This will remove sequences that contain none of these positional kmers.</p>
<p><code>-remove G:35:0,CNC:47:0</code><p>
<p> Remove sequences containing either G at position 35, or CNC at position 47. Sequences contain neither of these two kmers will be analyzed by PKA. </p>


      <h1><a class="mozTocH1" name="output"></a>Output</h1>

		<p>PKA outputs both sequence logo plots and tabular data file.</p>

      <h2><a class="mozTocH2" name="logo-plots"></a>Logo plots</h2>
      <h3><a class="mozTocH3" name="frequency-logo"></a>Frequency logo</h3>
      <p>In frequency logos, letters are rescaled relative to their frequency at each position, and then stacked on top of each other with more frequent letter on top of less frequent letter.</p>
      <h3><a class="mozTocH3" name="information-content-logo"></a>Information content logo</h3>
	<p>Similar to fequency logos, in information content logo (conventional sequence logo) letters are rescaled by frequency and stacked, and the total height of each position is further scaled relative to the information content, which is defined as </p>
<img src="./infocont.png">

<p>Here N is the number of letters in the alphabet (all possible letters that can occur at a position), and p<small>i</small> is the frequency of letter i at this position. </p> 
      <h3><a class="mozTocH3" name="probability-logo"></a>Probability logo</h3>
<p>Probability logo is first proposed and implemented in <a href="https://plogo.uconn.edu/">pLogo</a> for unweighted sequences, especially protein sequences. PKA extended the concept to ranked and weighted sequences. In probability logos, letters are rescaled relative to the statistical significance (-log10(p value)) of each letter at each position. Enriched letters stack on the top, whereas depleted letters stack on the bottom. Significant positions will have coordinates colored in red.</p>
      <h3><a class="mozTocH3" name="kmer-logo"></a>Kmer logo</h3>
	<p>PKA generates a new type of logo called kmer logo to visualize motifs in addition to single letter. Stacked letters at a position represents a single motif (the most significant) starting (or ending) at this position. The total height is rescaled relative to the significance of this motif. At each position, two motifs are shown, the most enriched on the top, and the most depletd on the bottom. Both read from top to bottom. Letters can be scaled by three different values: raw P value, Bonferroni corrected P values, or test statistics. In cases where P values are too small to be represented in the computer system (P < 1E-308 ), PKA will automatically switch to test statistics.  </p>
      <h3><a class="mozTocH3" name="fix"></a>Fixed positions</h3>
	<p>In many applications the input sequences may contain almost invariant letters at certain positions, such as GG in sgRNA target sequences (see example). The statistical model would not work well for extremely imbalanced samples, or the p-values are so small that it is hard to see signal at other positions. PKA allows user to fix positions where a single letter occurs in more than 75% of the sequences. This maximum frequency can be changed using <code>-fix</code> option. P-values for fixed positions will be ignored for probability logo, and the height of the fixed letter will be 1.1 times of the max total height of non-fixed position. </p>
	
      <h3><a class="mozTocH3" name="stackorder"></a>Stacking order</h3>
	<p>In all logo plots other than kmer logo, multiple letters can stack on top of each other. By default, letters with larger weights will be placed on top of letters with smaller weights. This can be reversed with <code>-stack_order -1</code>. To ignore weight information and use order specified by alphabet, use <code>-stack_order 0</code>.  </p>
	
	
      <h2><a class="mozTocH2" name="tabular-data-file"></a>Tabular data
        file</h2>
      <p>Regardless the type of input (unweighted/ranked/weighted), the
        first six columns of the output data file are always the same:</p>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Monaco;"><small>
          <ol>
            <li>kmer</li>
            <li>position: first nucleotide of the kmer in the sequence.</li>
            <li>shift</li>
            <li>statistics (positive for enrichment and negative for
              depletion)</li>
            <li>p.value (-log10)</li>
            <li>Bonferoni corrected p.value (-log10)</li>
          </ol>
        </small> </div>
      <p>Output from ranked input has no other columns other than the
        six above. Note the statistics and p-value in column 4 and 5 are
        based on normal approximation. See: </p>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Monaco;"><small>
          Carine A. Bellera et al, Normal Approximations to the
          Distributions of the Wilcoxon Statistics: Accurate to What N?
          Graphical Insights, <strong>Journal of
            Statistics Education</strong>, Volume 18, Number 2, (2010) <a
href="http://www.amstat.org/publications/jse/v18n2/bellera.pdf">Download PDF</a>
        </small></div>
      <p>Unranked input will generate four additional columns in the
        output:</p>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Monaco;"><small>
          <ol start="7">
            <li>observed fraction of input sequences have this kmer at
              this position allowing this amount of shift</li>
            <li>expected fraction of background sequences have this kmer
              at this position allowing this amount of shift</li>
            <li>ratio: column 7 divided by column 8</li>
            <li>column 7 divided by the average of other positions </li>
          </ol>
        </small> </div>
      <p></p>
      Weighted input will generate 6 more columns:
      <p></p>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Monaco;"><small>
          <ol start="7">
            <li>sample size for sample 1, i.e. the number of sequences
              have this kmer at this position when allowing this amount
              of shift. </li>
            <li>mean of weight for sample 1 sequences.</li>
            <li>standard deviation of weight for sample 1 sequences.</li>
            <li>sample size for sample 2, i.e. the number of sequences
              do not
              have this kmer at this position when allowing this amount
              of shift. </li>
            <li>mean of weight for sample 2 sequences.</li>
            <li>standard deviation of weight for sample 2 sequences.</li>
          </ol>
        </small> </div>
      <h1><a class="mozTocH1" name="examples"></a>Examples</h1>
		Here are two examples discussed in detail <font color=Red> (more details to be added)</font>. 
		<h2><a class="mozTocH2" name="example1"></a>Unranked: MicroRNA precursor / Drosha specificity </h2>
        <h4><a class="mozTocH4" name="example1-background"></a>Background </h4>

        <h4><a class="mozTocH4" name="example1-input"></a>Data set </h4>

        <h4><a class="mozTocH4" name="example1-setting"></a>Parameter setting </h4>

        <h4><a class="mozTocH4" name="example1-result"></a>Result </h4>

		<p> </p>
        <h2><a class="mozTocH2" name="example2"></a>Weighted/Ranked: CRISPR/Cas9 cutting efficiency </h2>
        <p> </p>
        <p> </p>

      <h1><a class="mozTocH1" name="command-line-options"></a>Command
        Line Options</h1>

      <p>Click <a href="commandline.usage">here</a> to view the command line options in plain text</p>
	  <p>An enhanced html version is comming soon.</p>

      <h1><a class="mozTocH1" name="copyright"></a>Copyright</h1>
      <p>Copyright (c) 2015 Xuebing Wu. </p>
      <p>PKA reuses the following library/code:<br>
        - <a href="http://www.boost.org/">Boost</a> library (version
        1.57.0)<br>
        - The <a href="http://digital.cs.usu.edu/%7Emjiang/ushuffle/">ushuffle</a>
        code written by Minghui Jiang <br>
      </p>
      <div style="border: 0px solid ; padding: 10px 40px; background:
        rgb(221, 221, 221) none repeat scroll 0% 50%;
        -moz-background-clip: -moz-initial; -moz-background-origin:
        -moz-initial; -moz-background-inline-policy: -moz-initial;
        font-family: Monaco;"><a>
          <p><b>License</b><br>
            <br>
            <small>
              Permission is hereby granted, free of charge, to any
              person obtaining a copy of this software and associated
              documentation files (the "Software"), to deal in the
              Software without restriction, including without limitation
              the rights to use, copy, modify, merge, publish,
              distribute, sublicense, and/or sell copies of the
              Software, and to permit persons to whom the Software is
              furnished to do so, subject to the following conditions:</small></p>
          <small>
            <p>The above copyright notice and this permission notice
              shall be included in all copies or substantial portions of
              the Software.</p>
          </small>
          <p><small>
              THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
              KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
              WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
              PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
              OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
              OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
              OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
              SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.</small></p>
        </a></div>
      <h1><a class="mozTocH1" name="author"></a>Author</h1>
      <p>PKA is developed and maintained by Xuebing Wu in the Bartel lab
        at Whitehead Institute.</p>
      <h1><a class="mozTocH1" name="acknowlegements"></a>Acknowledgements</h1>
      <p>I would like to thank Whitehead Institute IT system team (especially Paul McCabe, Andy Nutter-Upham, Craig Andrew, Alexan Mardigian) for help with the server setup. </p>
      <h1><a class="mozTocH1" name="reference"></a>Reference</h1>
      <p>coming soon</p>
    </div>
  </body>
</html>
